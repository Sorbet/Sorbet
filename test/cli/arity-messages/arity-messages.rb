# typed: true

def foo(x: nil); end
foo 1

def bar(x: nil, y: nil); end
bar(5, x: 8)

def baz(x: nil); end
baz 7
