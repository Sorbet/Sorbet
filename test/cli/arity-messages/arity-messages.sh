#!/bin/bash

set -eoc

main/sorbet --silence-dev-message test/cli/arity-messages/arity-messages.rb 2>&1
