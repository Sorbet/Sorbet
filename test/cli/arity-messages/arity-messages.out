test/cli/arity-messages/arity-messages.rb:4: Too many positional arguments provided for method `Object#foo`. Expected: `0`, got: `1` https://sorbet.org/docs/error-reference#7004
     4 |foo 1
        ^^^^^
    test/cli/arity-messages/arity-messages.rb:3: `foo` defined here
     3 |def foo(x: nil); end
        ^^^^^^^^^^^^^^^
    test/cli/arity-messages/arity-messages.rb:4: `Object#foo` has optional keyword arguments. Did you mean to provide a value for `x`?
     4 |foo 1
        ^^^^^

test/cli/arity-messages/arity-messages.rb:5: Unrecognized keyword argument `y` passed for method `Object#foo` https://sorbet.org/docs/error-reference#7004
     5 |foo(y: 1)
        ^^^^^^^^^

test/cli/arity-messages/arity-messages.rb:8: Too many positional arguments provided for method `Object#bar`. Expected: `0`, got: `1` https://sorbet.org/docs/error-reference#7004
     8 |bar(5, x: 8)
        ^^^^^^^^^^^^
    test/cli/arity-messages/arity-messages.rb:7: `bar` defined here
     7 |def bar(x: nil, y: nil); end
        ^^^^^^^^^^^^^^^^^^^^^^^
    test/cli/arity-messages/arity-messages.rb:8: `Object#bar` has optional keyword arguments. Did you mean to provide a value for `y`?
     8 |bar(5, x: 8)
        ^^^^^^^^^^^^

test/cli/arity-messages/arity-messages.rb:9: Too many positional arguments provided for method `Object#bar`. Expected: `0`, got: `2` https://sorbet.org/docs/error-reference#7004
     9 |bar(5, 8)
        ^^^^^^^^^
    test/cli/arity-messages/arity-messages.rb:7: `bar` defined here
     7 |def bar(x: nil, y: nil); end
        ^^^^^^^^^^^^^^^^^^^^^^^
    test/cli/arity-messages/arity-messages.rb:9: `Object#bar` has optional keyword arguments. Did you mean to provide a value for `x`?
     9 |bar(5, 8)
        ^^^^^^^^^

test/cli/arity-messages/arity-messages.rb:12: Too many positional arguments provided for method `Object#baz`. Expected: `0`, got: `1` https://sorbet.org/docs/error-reference#7004
    12 |baz 7
        ^^^^^
    test/cli/arity-messages/arity-messages.rb:11: `baz` defined here
    11 |def baz(x: nil); end
        ^^^^^^^^^^^^^^^
    test/cli/arity-messages/arity-messages.rb:12: `Object#baz` has optional keyword arguments. Did you mean to provide a value for `x`?
    12 |baz 7
        ^^^^^

test/cli/arity-messages/arity-messages.rb:15: Not enough arguments provided for method `Object#quux`. Expected: `3`, got: `2` https://sorbet.org/docs/error-reference#7004
    15 |quux(2, z: 3)
        ^^^^^^^^^^^^^
    test/cli/arity-messages/arity-messages.rb:14: `Object#quux` defined here
    14 |def quux(x, y, z); end
        ^^^^^^^^^^^^^^^^^

test/cli/arity-messages/arity-messages.rb:18: Unrecognized keyword argument `c` passed for method `Object#qaadr` https://sorbet.org/docs/error-reference#7004
    18 |qaadr({a: 1}, {b: 2}, {c: 3})
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

test/cli/arity-messages/arity-messages.rb:18: Too many positional arguments provided for method `Object#qaadr`. Expected: `0`, got: `0` https://sorbet.org/docs/error-reference#7004
    18 |qaadr({a: 1}, {b: 2}, {c: 3})
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    test/cli/arity-messages/arity-messages.rb:17: `qaadr` defined here
    17 |def qaadr(e: 1); end
        ^^^^^^^^^^^^^^^
    test/cli/arity-messages/arity-messages.rb:18: `Object#qaadr` has optional keyword arguments. Did you mean to provide a value for `e`?
    18 |qaadr({a: 1}, {b: 2}, {c: 3})
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

test/cli/arity-messages/arity-messages.rb:21: Too many positional arguments provided for method `Object#pippo`. Expected: `1`, got: `2` https://sorbet.org/docs/error-reference#7004
    21 |pippo(2, 3, z: 4)
        ^^^^^^^^^^^^^^^^^
    test/cli/arity-messages/arity-messages.rb:20: `pippo` defined here
    20 |def pippo(x, **kwargs); end
        ^^^^^^^^^^^^^^^^^^^^^^

test/cli/arity-messages/arity-messages.rb:24: Not enough arguments provided for method `Object#pluto`. Expected: `1+`, got: `0` https://sorbet.org/docs/error-reference#7004
    24 |pluto
        ^^^^^
    test/cli/arity-messages/arity-messages.rb:23: `Object#pluto` defined here
    23 |def pluto(x, *args); end
        ^^^^^^^^^^^^^^^^^^^

test/cli/arity-messages/arity-messages.rb:27: Too many positional arguments provided for method `Object#paperino`. Expected: `1`, got: `2` https://sorbet.org/docs/error-reference#7004
    27 |paperino(2, 3, z: 4)
        ^^^^^^^^^^^^^^^^^^^^
    test/cli/arity-messages/arity-messages.rb:26: `paperino` defined here
    26 |def paperino(x, y: nil, **kwargs); end
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    test/cli/arity-messages/arity-messages.rb:27: `Object#paperino` has optional keyword arguments. Did you mean to provide a value for `y`?
    27 |paperino(2, 3, z: 4)
        ^^^^^^^^^^^^^^^^^^^^
Errors: 11
